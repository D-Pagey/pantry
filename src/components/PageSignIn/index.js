import React, { useContext } from 'react';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import { Redirect } from 'react-router-dom';
import { firebase } from '../../services';
import { FirebaseContext } from '../ProviderFirebase';

const PageSignIn = () => {
    const { isAuthed, setUser } = useContext(FirebaseContext);

    // Configure FirebaseUI.
    const uiConfig = {
        // Popup signin flow rather than redirect flow.
        signInFlow: 'popup',
        // Redirect to /signedIn after sign in is successful. Alternatively you can provide a callbacks.signInSuccess function.
        // signInSuccessUrl: '/signedIn',
        callbacks: {
            // Avoid redirects after sign-in.
            signInSuccessWithAuthResult: (result) =>
                setUser({ name: result.user.displayName, email: result.user.email })
        },
        // We will display Google and Facebook as auth providers.
        signInOptions: [firebase.auth.GoogleAuthProvider.PROVIDER_ID]
    };

    if (isAuthed) {
        return <Redirect to="/" />;
    }

    return (
        <div data-testid="pageSignIn">
            <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
        </div>
    );
};

export default PageSignIn;
