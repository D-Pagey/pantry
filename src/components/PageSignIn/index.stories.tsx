import { PageSignIn } from '.';

export default { title: 'PageSignIn' };

export const normal = (): JSX.Element => <PageSignIn />;
