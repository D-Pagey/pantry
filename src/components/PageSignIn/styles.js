import styled from 'styled-components';

export const Wrapper = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    padding: 1rem;
`;

export const Title = styled.h1`
    font-weight: normal;
    margin: 1rem 0;
`;

export const Image = styled.img`
    margin: 2rem 0 0;
    width: 400px;
`;

export const EmailWrapper = styled.div`
    display: flex;
    margin: 0;
    width: 330px;
`;
