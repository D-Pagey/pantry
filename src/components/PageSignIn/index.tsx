import React, { useContext, useState } from 'react';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import { Redirect } from 'react-router-dom';

import { firebase } from '../../services';
import { AuthContext } from '../ProviderAuth';
import { Layout } from '../Layout';
import { Input } from '../Input';
import { Button } from '../Button';
import loginImage from './assets/mobile-login.svg';
import * as S from './styles';

export const PageSignIn = () => {
    const [email, setEmail] = useState('');
    const { isAuthed, setUser } = useContext(AuthContext);

    const handleEmailChange = (event: any) => setEmail(event.target.value);

    // Configure FirebaseUI.
    const uiConfig = {
        // Popup signin flow rather than redirect flow.
        signInFlow: 'popup',
        // Redirect to /signedIn after sign in is successful. Alternatively you can provide a callbacks.signInSuccess function.
        // signInSuccessUrl: '/signedIn',
        callbacks: {
            // Avoid redirects after sign-in.
            signInSuccessWithAuthResult: (result: any) => {
                setUser({ name: result.user.displayName, email: result.user.email, photo: result.user.photoURL });
                return false;
            }
        },
        // We will display Google and Facebook as auth providers.
        signInOptions: [firebase.auth.GoogleAuthProvider.PROVIDER_ID]
    };

    if (isAuthed) {
        return <Redirect to="/food" />;
    }

    return (
        <Layout title="Sign in">
            <S.Wrapper>
                <S.Title>Who are you?</S.Title>

                <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />

                <p>or</p>

                <S.EmailWrapper>
                    <Input
                        margin="0 2rem 0 0"
                        onChange={handleEmailChange}
                        placeholder="Email me a magic link"
                        value={email}
                    />
                    <Button>Send</Button>
                </S.EmailWrapper>

                <S.Image src={loginImage} alt="login" />
            </S.Wrapper>
        </Layout>
    );
};
