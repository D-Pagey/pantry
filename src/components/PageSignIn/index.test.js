import React from 'react';
import { Redirect } from 'react-router-dom';
import PageSignIn from '.';

jest.mock('react-firebaseui/StyledFirebaseAuth');

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Redirect: jest.fn(() => null)
}));

const context = {
    isAuthed: false,
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    setUser: () => {}
};

describe('PageSignIn component', () => {
    it('should render', () => {
        const { container } = render(<PageSignIn />, context);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should redirect if authed', () => {
        render(<PageSignIn />, { ...context, isAuthed: true });
        expect(Redirect).toHaveBeenCalledWith({ to: '/' }, expect.any(Object));
    });
});
