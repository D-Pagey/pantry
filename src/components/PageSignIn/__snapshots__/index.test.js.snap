// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSignIn component should render 1`] = `
<div
  data-testid="pageSignIn"
/>
`;
